packages/lib/RotatingLogs.test.ts

```diff
@@ -0,0 +1,51 @@
+import { open, close, write, readdir, rm } from 'fs-extra';
+import RotatingLogs from './RotatingLogs';
+import { createTempDir } from './testing/test-utils';
+
+jest.useFakeTimers();
+
+const createEmptyFileOfSize = async (fileName: string, size: number) => {
+	const fd = await open(fileName, 'w');
+	await write(fd, 'test', size);
+	await close(fd);
+};
+
+describe('RotatingLogs', () => {
+	let dir: string;
+	let files: string[];
+	let rotatingLogs: RotatingLogs;
+
+	beforeEach(async () => {
+		dir = await createTempDir();
+		rotatingLogs = new RotatingLogs(dir, 1024, 1000);
+		await createEmptyFileOfSize(`${dir}/log.txt`, 1024);
+	});
+
+	afterEach(async () => {
+		files = [];
+		await rm(dir, { recursive: true });
+	});
+
+	test('should exists an log.txt file inside the folder', async () => {
+		files = await readdir(dir);
+		expect(files.find(file => file.match(/^log.txt$/gi))).toBeTruthy();
+	});
+
+	test('should rename log.txt to log-TIMESTAMP.txt', async () => {
+		await rotatingLogs.cleanActiveLogFile();
+		files = await readdir(dir);
+		expect(files.find(file => file.match(/^log.txt$/gi))).toBeFalsy();
+		expect(files.find(file => file.match(/^log-[0-9]+.txt$/gi))).toBeTruthy();
+	});
+
+	test('should delete inactive log file older than 1 second', async () => {
+		await rotatingLogs.cleanActiveLogFile();
+		files = await readdir(dir);
+		expect(files.find(file => file.match(/^log-[0-9]+.txt$/gi))).toBeTruthy();
+		setTimeout(async () => {
+			await rotatingLogs.deleteNonActiveLogFiles();
+			files = await readdir(dir);
+			expect(files.find(file => file.match(/^log-[0-9]+.txt$/gi))).toBeFalsy();
+		}, 1000);
```

Comment:
This can't work. The test is going to exit while your timer is still running. Since you can now set inactiveMaxAge you don't need the fake timers anymore.

You can set it to 1 ms. Then create a file, wait for 2 ms using `await msleep(2)`, and then run your deleteNonActiveFile function. Check that the file has indeed be deleted